### PR TITLE
Add Ganglion firmware toggle for Mac users

### PR DESCRIPTION
**Pull request by retiutut**
Temporary fix for #1191

### CODE REVIEW

**Comment by retiutut** [comment]
image

File: OpenBCI_GUI/ControlPanel.pde [hunk 1]
```diff
@@ -691,6 +695,17 @@ class BLEBox {
 
         createRefreshBLEButton("refreshGanglionBLEButton", "START SEARCH", x + padding, y + padding*4 + 72 + 8, w - padding*5, 24);
         createGanglionBLEMenuList(bleBox_cp5, "bleList", x + padding, y + padding*3 + 8, w - padding*2, 72, p3);
+        
+        //Temporary fix for Mac OS X and Ganglion BLE
+        if (isMac() || isWindows()) {
```
Comment (philippitts) [inline OpenBCI_GUI/ControlPanel.pde:700]: Why is this checking if it's Windows? This should only apply to MacOS.

File: OpenBCI_GUI/OpenBCI_GUI.pde [hunk 2]
```diff
@@ -312,6 +312,8 @@ boolean isFFTFiltered = true; //yes by default ... this is used in dataProcessin
 StringBuilder globalScreenResolution;
 StringBuilder globalScreenDPI;
 
+boolean usingGanglionV3andMacOS = true;
```
Comment (philippitts) [inline OpenBCI_GUI/OpenBCI_GUI.pde:315]: I think this should be false by default.

File: OpenBCI_GUI/ControlPanel.pde (hunk 1 above)
Comment (philippitts) [reply OpenBCI_GUI/ControlPanel.pde:700]: I see the buttons (but shouldn't) on Windows.
image

**Review by philippitts — CHANGES_REQUESTED** [review]

File: OpenBCI_GUI/OpenBCI_GUI.pde (hunk 2 above)
Comment (retiutut) [reply OpenBCI_GUI/OpenBCI_GUI.pde:315]: Ok

**Comment by retiutut** [comment]
Runs as expected on Mac OS 13 with Apple Silicon (Arm64) mac.

Though I saw some serious issues with packet loss using Ganglion v3.0.1, BrainFlow 5.10.0, and Native Bluetooth. @philippitts

**Comment by retiutut** [comment]
```
=================================================
||             INITIALIZING SYSTEM             ||
=================================================

MAC address for Ganglion is 3C89F9CC-86F7-F9AB-210F-112455EE343D
[DEFAULT]: Detected Ganglion firmware version 3
Sending config string to board: n
2023-09-20 15:56:08.466 java[13033:776969] Peripheral ready to send: <CBPeripheral: 0x7fe1649b1890, identifier = 3C89F9CC-86F7-F9AB-210F-112455EE343D, name = Simblee, mtu = 23, state = connected>
OpenBCI_GUI: Channel count set to 4
opening socket failed!
        > address:127.0.0.1, port:12350 [group:null]
        > Address already in use
[SUCCESS]: Marker Widget: Listening for markers on 127.0.0.1:12350
InitSettings: Saving Default Settings to file!
SessionSettings: 4 active widgets saved!
[SUCCESS]: Session started!
[DEFAULT]: openBCI_GUI: startButton was pressed. Starting data transfer, wait a few seconds.
OpenBCI_GUI: openNewLogFile: opened ODF output file: /Users/richard/Documents/OpenBCI_GUI/Recordings/OpenBCISession_2023-09-20_15-55-46/OpenBCI-RAW-2023-09-20_15-56-10.txt
OpenBCI_GUI: BrainFlow Streamer Location: file:///Users/richard/Documents/OpenBCI_GUI/Recordings/OpenBCISession_2023-09-20_15-55-46/BrainFlow-RAW_2023-09-20_15-55-46_0.csv:w
Brainflow start streaming
[DEFAULT]: Data stream started.
2023-09-20 15:56:10.187 java[13033:776970] Peripheral ready to send: <CBPeripheral: 0x7fe1649b1890, identifier = 3C89F9CC-86F7-F9AB-210F-112455EE343D, name = Simblee, mtu = 23, state = connected>
WARNING: Lost 1 Samples Between 0-1
[WARN]: Lost packets detected, open packet loss widget for more info
[INFO]: Data streaming is running as usual
WARNING: Lost 1 Samples Between 0-1
WARNING: Lost 1 Samples Between 0-1
WARNING: Lost 1 Samples Between 0-1
WARNING: Lost 1 Samples Between 0-1
WARNING: Lost 1 Samples Between 0-1
WARNING: Lost 1 Samples Between 0-1
WARNING: Lost 1 Samples Between 0-1
WARNING: Lost 1 Samples Between 0-1
WARNING: Lost 1 Samples Between 0-1
WARNING: Lost 1 Samples Between 0-1
WARNING: Lost 1 Samples Between 0-1
WARNING: Lost 1 Samples Between 0-1
WARNING: Lost 1 Samples Between 0-1
WARNING: Lost 1 Samples Between 0-1
WARNING: Lost 1 Samples Between 0-1
WARNING: Lost 1 Samples Between 0-1
WARNING: Lost 1 Samples Between 0-1
WARNING: Lost 1 Samples Between 0-1
WARNING: Lost 1 Samples Between 0-1
[DEFAULT]: openBCI_GUI: stopButton was pressed. Stopping data transfer, wait a few seconds.
Brainflow stop streaming
[DEFAULT]: Data stream stopped.
```

**Comment by retiutut** [comment]
Screenshot 2023-09-20 at 4 18 18 PM

**Comment by philippitts** [comment]
There was an issue with the Ganglion v3.0.1 build posted on the releases page as discussed [here](https://openbci.com/forum/index.php?p=/discussion/3686/compatibility-issue-with-bioera-and-ganglion-firmware-v3-0#latest). Can you try re-downloading and flashing the Ganglion v3.0.1 firmware?

**Pull request closed without merging**